stan/math/prim/mat/prob/ordered_logistic_lpmf.hpp

```diff
@@ -112,126 +118,67 @@ typename return_type<T_loc, T_cut>::type ordered_logistic_lpmf(
-    const std::vector<int>& y,
-    const Eigen::Matrix<T_loc, Eigen::Dynamic, 1>& lambda,
-    const Eigen::Matrix<T_cut, Eigen::Dynamic, 1>& c) {
-  return ordered_logistic_lpmf<false>(y, lambda, c);
-}
+  int K = c_vec[0].size() + 1;
+  int N = length(lambda);
 
-/**
- * Returns the (natural) log probability of the specified array
- * of integers given the vector of continuous locations and
- * array of specified cutpoints in an ordered logistic model.
- *
- * <p>Typically the continous location
- * will be the dot product of a vector of regression coefficients
- * and a vector of predictors for the outcome.
- *
- * @tparam propto True if calculating up to a proportion.
- * @tparam T_y Type of y variable (should be std::vector<int>).
- * @tparam T_loc Location type.
- * @tparam T_cut Cut-point type.
- * @param y Array of integers
- * @param lambda Vector of continuous location variables.
- * @param c array of Positive increasing vectors of cutpoints.
- * @return Log probability of outcome given location and
- * cutpoints.
- * @throw std::domain_error If the outcome is not between 1 and
- * the number of cutpoints plus 2; if the cutpoint vector is
- * empty; if the cutpoint vector contains a non-positive,
- * non-finite value; or if the cutpoint vector is not sorted in
- * ascending order.
- * @throw std::invalid_argument If y and lambda are different
- * lengths, or if y and the array of cutpoints are of different
- * lengths.
- */
-template <bool propto, typename T_loc, typename T_cut>
-typename return_type<T_loc, T_cut>::type ordered_logistic_lpmf(
-    const std::vector<int>& y,
-    const Eigen::Matrix<T_loc, Eigen::Dynamic, 1>& lambda,
-    const std::vector<Eigen::Matrix<T_cut, Eigen::Dynamic, 1> >& c) {
-  typename return_type<T_loc, T_cut>::type logp_n(0.0);
+  check_consistent_sizes(function, "Integers", y, "Locations", lambda);
 
-  using std::exp;
-  using std::log;
+  for (int n = 0; n < N; n++) {
+    check_bounded(function, "Random variable", y_vec[n], 1, K);
+    check_finite(function, "Location parameter", lam_vec[n]);
+  }
 
-  static const char* function = "ordered_logistic";
+  for (size_t i = 0, size_ = length(c_vec); i < size_; i++) {
+    check_ordered(function, "Cut-points", c_vec[i]);
+    check_greater(function, "Size of cut points parameter", c_vec[i].size(), 0);
+    check_finite(function, "Final cut-point", c_vec[i](c_vec[i].size() - 1));
+    check_finite(function, "First cut-point", c_vec[i](0));
+  }
 
-  int N = lambda.size();
+  ordLog_helper<T_partials_return> ordhelp;
+  operands_and_partials<T_loc, T_cut> ops_partials(lambda, c);
 
-  check_consistent_sizes(function, "Integers", y, "Locations", lambda);
-  check_consistent_sizes(function, "Integers", y, "Cut-points", c);
+  T_partials_return logp(0.0);
 
-  for (int i = 0; i < N; ++i) {
-    int K = c[i].size() + 1;
+  for (int n = 0; n < N; ++n) {
+    for (size_t i = 0, size_ = length(c_vec); i < size_; i++) {
```

Comment:
Vectorization in the _lpdfs works like:

```
vector[2] a;
vector[2] b;

target += my_lpmf(a, b)
```

is the same as:

```
target += my_lpmf(a[0], b[0])
target += my_lpmf(a[1], b[1])
```

This is doing something like:
```
target += my_lpmf(a[0], b[0])
target += my_lpmf(a[0], b[1])
target += my_lpmf(a[1], b[0])
target += my_lpmf(a[1], b[1])
```